layer.py

```diff
@@ -339,7 +339,7 @@ def name(self):
 
     @property
     def metadata(self) -> Optional[QgsProviderMetadata]:
-        if not self.layer.isValid():
+        if not self.layer.isValid() or not self.layer.dataProvider():
```

Comment:
Is there a situation where `QgsMapLayer::dataProvider()` returns `nullptr` and `QgsMapLayer::isValid()` returns `True`? Not 100% sure if the second part after the `or` is needed?

Reply:
@suricactus , yes, vector tiles layers don't have a provider.

Reply:
Good to know, there are other other checks in the codebases that check only if the layer is valid. Thanks!